library/std/src/io/impls.rs

```diff
@@ -287,6 +287,9 @@ impl Read for &[u8] {
     #[inline]
     fn read_exact(&mut self, buf: &mut [u8]) -> io::Result<()> {
         if buf.len() > self.len() {
+            // `read_exact` makes no promise about the content of `buf` if it
+            // fails so don't bother about that.
+            *self = &self[self.len()..];
```

Comment:
doct... actually I'm pretty sure there isn't a useful doctest for "lol no promises", right? don't worry about this one, if so.